search/geocoder.cpp

```diff
@@ -434,6 +440,32 @@ void Geocoder::GoInViewport()
   GoImpl(infos, true /* inViewport */);
 }
 
+void Geocoder::ClearCaches()
+{
+  m_pivotRectsCache.Clear();
+  m_localityRectsCache.Clear();
+
+  m_matchersCache.clear();
+  m_streetsCache.Clear();
+  m_hotelsCache.Clear();
+  m_hotelsFilter.ClearCaches();
+  m_postcodes.Clear();
+}
+
+void Geocoder::SetParamsForCategorialSearch(Params const & params)
+{
+  m_params = params;
```

Comment:
This code duplicates part of code from SetParams, could you please get rid of this duplication?

Reply:
I couldn't find a neat way to do it. If you insist, I can factor out those 4 lines to a helper method.
Note that the log message must be duplicated even if it is the same (the log prefix contains the call site).